src/crystal/datum.cr

```diff
@@ -144,13 +155,18 @@ module Crystal
       @raw.pretty_print(pp)
     end
 
+    # :nodoc:
+    def to_json(builder : JSON::Builder) : Nil
+      @raw.to_json builder
+    end
```

Comment:
`Log` shouldn't depend on `JSON`. I'm pretty sure we don't already. Does `require "log"` in an empty file work after this?

Reply:
It doesn't depend on JSON as long as this method is not called. This is the same as `Object#to_json`.

Reply:
I thought that the type restriction would cause a parse error but I guess not

Reply:
It's done like that so you don't have to use forward definitions. Then these types are only checked when called.

That said, I wouldn't discard that in the future these will be checked after all types have been defined to catch errors earlier. And that will mean this will break.

So I suggest putting this into `require "log/json"`.

Reply:
In that case, should I move these methods into `log/context.cr` versus `datum`?  Really these methods are useable in anything datum would be used in.

Reply:
At the current state of things this feels unjustified. You don't need to `require "object/json"` to call `to_json` on any object. Maybe in the future it will be required. But we don't actually know how this could play out. Maybe we'll end up with an entirely different solution.
So I'd not force an additional require just to use a simple delegation method.

Reply:
You need to `require "json"`. `object.to_json` is otherwise not defined

Reply:
Yes, but but it only works if the actual type implements `#to_json(json : JSON::Builder)`.

Reply:
Let's just leave this as-is now.

Either this changes before 1.0, or it changed in 2.0, because it'd be a massive breaking change anyway. Neither of these possibilities affect this PR.

Reply:
It is not, but the files are parsed. 

Also it is not true that datum values needs to be JSON compatible. 

I think the`Log::Context#to_json` should be in a `log/json.cr` file

Reply:
Maybe something in the middle and just move it to `Log::Context`, so it's specific just to that?  I'm not so sure needing to require another file is really necessary.